public/app/features/plugins/built_in_plugins.ts

```diff
@@ -73,10 +73,25 @@ const canvasPanel = async () => await import(/* webpackChunkName: "canvasPanel"
 const iconPanel = async () => await import(/* webpackChunkName: "iconPanel" */ 'app/plugins/panel/icon/module');
 const graphPanel = async () => await import(/* webpackChunkName: "graphPlugin" */ 'app/plugins/panel/graph/module');
 const heatmapPanel = async () =>
-  await import(/* webpackChunkName: "heatmapPlugin" */ 'app/plugins/panel/heatmap/module');
+  await import(/* webpackChunkName: "heatmapPanel" */ 'app/plugins/panel/heatmap/module');
+const heatmapPanelOLD = async () =>
+  await import(/* webpackChunkName: "heatmapPanelOLD" */ 'app/plugins/panel/heatmap-old/module');
+
 const tableOldPanel = async () =>
   await import(/* webpackChunkName: "tableOldPlugin" */ 'app/plugins/panel/table-old/module');
 
+// Automatically migrate heatmap panel.
+const useLegacyHeatmapPanel =
+  config.featureToggles.useLegacyHeatmapPanel || window.location?.search?.includes('useLegacyHeatmapPanel=true');
```

Comment:
rather than have custom url handling here... we could do it consistently with something like:  https://github.com/grafana/grafana/pull/50273/files#diff-8e6d1dc8bffebb9fc6d982a51a284e963b36aefdd18a58451fff8abb47068db5R158